handlers/upload.go

```diff
@@ -82,15 +82,17 @@ func (r UploadVODRequest) getTargetHlsOutput() UploadVODRequestOutputLocation {
 	return UploadVODRequestOutputLocation{}
 }
 
-func (r UploadVODRequest) getTargetMp4Output() (UploadVODRequestOutputLocation, bool) {
+func (r UploadVODRequest) getTargetMp4Output() (UploadVODRequestOutputLocation, bool, bool) {
```

Comment:
@leszko need your input on commit `3f9f5eb4013b2174ac76f088bba9e3788e6408eb`.  The `MP4` field in  `UploadVODRequestOutputLocation` is already set by studio when we need to generate an mp4 file. The change here adds an option to generate a regular old mp4 (MP4 set to `enabled`) or a fragmented mp4 (MP4 set to `fragmented`). In other words, we should only generate either a regular mp4 or fragmented-mp4. Do you see any issues or dependencies from Studio perspective?

Reply:
I don't love this - that parameter is for controlling _whether_ an MP4 is generated, rather than the type. Could we add the same parameter but for fMP4? That also opens the door to generating both, if that's ever something customers want

Reply:
Will update.

Reply:
Added new parameter for fMP4 which required a refactor - see latest commits (still wip but commit `f467915c364a0032847dec5f1445d2316570d29f` should be good for review).